homeassistant/components/homematicip_cloud.py

```diff
@@ -0,0 +1,170 @@
+    vol.Optional(DOMAIN): [vol.Schema({
+        vol.Optional(CONF_NAME, default=''): cv.string,
+        vol.Required(CONF_ACCESSPOINT): cv.string,
+        vol.Required(CONF_AUTHTOKEN): cv.string,
+    })],
+}, extra=vol.ALLOW_EXTRA)
+
+EVENT_HOME_CHANGED = 'homematicip_home_changed'
+EVENT_DEVICE_CHANGED = 'homematicip_device_changed'
+EVENT_GROUP_CHANGED = 'homematicip_group_changed'
+EVENT_SECURITY_CHANGED = 'homematicip_security_changed'
+EVENT_JOURNAL_CHANGED = 'homematicip_journal_changed'
+
+ATTR_HOME_ID = 'home_id'
+ATTR_HOME_LABEL = 'home_label'
+ATTR_DEVICE_ID = 'device_id'
+ATTR_DEVICE_LABEL = 'device_label'
+ATTR_STATUS_UPDATE = 'status_update'
+ATTR_FIRMWARE_STATE = 'firmware_state'
+ATTR_LOW_BATTERY = 'low_battery'
+ATTR_SABOTAGE = 'sabotage'
+ATTR_RSSI = 'rssi'
+
+
+def setup(hass, config):
+    """Set up the HomematicIP component."""
+    # pylint: disable=import-error, no-name-in-module
+    from homematicip.home import Home
+    hass.data.setdefault(DOMAIN, {})
+    homes = hass.data[DOMAIN]
+    accesspoints = config.get(DOMAIN, [])
+
+    def _update_event(events):
+        """Handle incoming HomeMaticIP events."""
+        for event in events:
+            etype = event['eventType']
+            edata = event['data']
+            if etype == 'DEVICE_CHANGED':
+                dispatcher_send(hass, EVENT_DEVICE_CHANGED, edata.id)
+            elif etype == 'GROUP_CHANGED':
+                dispatcher_send(hass, EVENT_GROUP_CHANGED, edata.id)
+            elif etype == 'HOME_CHANGED':
+                dispatcher_send(hass, EVENT_HOME_CHANGED, edata.id)
+            elif etype == 'JOURNAL_CHANGED':
+                dispatcher_send(hass, EVENT_SECURITY_CHANGED, edata.id)
+        return True
+
+    for device in accesspoints:
+        name = device.get(CONF_NAME)
+        accesspoint = device.get(CONF_ACCESSPOINT)
+        authtoken = device.get(CONF_AUTHTOKEN)
+
+        home = Home()
+        if name.lower() == 'none':
+            name = ''
+        home.label = name
+        try:
+            home.set_auth_token(authtoken)
+            home.init(accesspoint)
+            if home.get_current_state():
+                _LOGGER.info("Connection to HMIP established")
+            else:
+                _LOGGER.warning("Connection to HMIP could not be established")
+                return False
+        except timeout:
+            _LOGGER.warning("Connection to HMIP could not be established")
+            return False
+        homes[home.id] = home
+        home.onEvent += _update_event
+        home.enable_events()
+        _LOGGER.info('HUB name: %s, id: %s', home.label, home.id)
+
+        for component in ['sensor']:
+            load_platform(hass, component, DOMAIN,
+                          {'homeid': home.id}, config)
+    return True
+
+
+class HomematicipGenericDevice(Entity):
+    """Representation of an HomematicIP generic device."""
+
+    def __init__(self, hass, home, device, signal=None):
```

Comment:
Don't pass in `hass`. It will be set on the entity when it has been added to home assistant.

Reply:
`signal` parameter doesn't seem to be used.